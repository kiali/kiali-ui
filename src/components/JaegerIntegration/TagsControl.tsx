import * as React from 'react';
import { Button, Form, FormGroup, Popover, TextInput } from '@patternfly/react-core';
import { InfoAltIcon } from '@patternfly/react-icons';
import { style } from 'typestyle';

interface TagsControlProps {
  disable?: boolean;
  tags?: string;
  onChange: (value: string) => void;
}

const tagsInput = style({ marginLeft: '-100px' });

export class TagsControl extends React.PureComponent<TagsControlProps, {}> {
  tagsHelp = () => {
    return (
      <>
        <Popover
          position="right"
          className={tagsInput}
          bodyContent={
            <>
              Values should be in the{' '}
              <a rel="noopener noreferrer" href="https://brandur.org/logfmt" target="_blank">
                logfmt
              </a>{' '}
              format.
              <ul>
                <li>Use space for conjunctions</li>
                <li>Values containing whitespace should be enclosed in quotes</li>
              </ul>
              <code>error=true db.statement="select * from User"</code>
            </>
          }
        >
          <>
            <Button variant="plain">
              <InfoAltIcon />
            </Button>
            e.g. http.status_code=200 error=true
          </>
        </Popover>
      </>
    );
  };

  render() {
    const { tags } = this.props;
    return (
      <Form isHorizontal={true}>
        <FormGroup label="Tags" fieldId="jaeger-tags" helperText={this.tagsHelp()}>
          <TextInput
            value={tags}
            type="text"
            onChange={this.props.onChange}
            aria-label="tagsJaegerTraces"
            className={tagsInput}
          />
        </FormGroup>
      </Form>
    );
  }
}

export default TagsControl;
