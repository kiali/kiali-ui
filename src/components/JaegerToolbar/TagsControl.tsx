import * as React from 'react';
import { connect } from 'react-redux';
import { Col, Form, Card, FormGroup, FormControl, FieldLevelHelp } from 'patternfly-react';
import { KialiAppState } from '../../store/Store';

interface TagsControlProps {
  fetching: boolean;
  tags: string;
  onChange: (event: any) => void;
}

export class TagsControl extends React.PureComponent<TagsControlProps, {}> {
  constructor(props: TagsControlProps) {
    super(props);
  }

  tagsHelp = () => {
    return (
      <Card>
        <Card.Title>
          Values should be in the{' '}
          <a href="https://brandur.org/logfmt" target="_blank">
            logfmt
          </a>{' '}
          format.
        </Card.Title>
        <Card.Body>
          <ul>
            <li>Use space for conjunctions</li>
            <li>Values containing whitespace should be enclosed in quotes</li>
          </ul>
        </Card.Body>
        <Card.Footer>
          <code>error=true db.statement="select * from User"</code>
        </Card.Footer>
      </Card>
    );
  };

  render() {
    const { fetching, tags } = this.props;
    return (
      <FormGroup style={{ display: 'inline-flex' }}>
        <Col componentClass={Form.ControlLabel}>
          Tags
          <FieldLevelHelp content={this.tagsHelp()} placement={'bottom'} />
        </Col>
        <FormControl
          type="text"
          disabled={fetching}
          defaultValue={tags}
          placeholder={'e.g. http.status_code=200 error=true'}
          style={{ width: '400px', marginLeft: '10px' }}
          onChange={e => this.props.onChange(e)}
        />
      </FormGroup>
    );
  }
}

const mapStateToProps = (state: KialiAppState) => {
  return {
    fetching: state.jaegerState.search.serviceSelected === '',
    tags: state.jaegerState.search.tags
  };
};

const TagsControlContainer = connect(mapStateToProps)(TagsControl);

export default TagsControlContainer;
