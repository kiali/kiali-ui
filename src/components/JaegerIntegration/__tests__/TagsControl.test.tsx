import * as React from 'react';
import { shallow } from 'enzyme';
import { TagsControl } from '../TagsControl';
import { shallowToJson } from 'enzyme-to-json';

describe('TagsControls', () => {
  let wrapper, onChangeMock;
  beforeEach(() => {
    onChangeMock = jest.fn();
    wrapper = shallow(<TagsControl onChange={onChangeMock} disable={false} tags={''} />);
  });

  it('renders TagsControl correctly', () => {
    expect(shallowToJson(wrapper)).toBeDefined();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
