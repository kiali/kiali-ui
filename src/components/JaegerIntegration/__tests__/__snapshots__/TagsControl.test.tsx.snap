// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsControls renders TagsControl correctly 1`] = `
<Form
  isHorizontal={true}
>
  <FormGroup
    fieldId="jaeger-tags"
    helperText={
      <React.Fragment>
        <Popover
          appendTo={[Function]}
          aria-label=""
          bodyContent={
            <React.Fragment>
              Values should be in the
               
              <a
                href="https://brandur.org/logfmt"
                rel="noopener noreferrer"
                target="_blank"
              >
                logfmt
              </a>
               
              format.
              <ul>
                <li>
                  Use space for conjunctions
                </li>
                <li>
                  Values containing whitespace should be enclosed in quotes
                </li>
              </ul>
              <code>
                error=true db.statement="select * from User"
              </code>
            </React.Fragment>
          }
          className="f6vm0mb"
          closeBtnAriaLabel="Close"
          enableFlip={true}
          footerContent={null}
          headerContent={null}
          hideOnOutsideClick={true}
          isVisible={null}
          maxWidth="calc(1.5rem + 18.75rem)"
          onHidden={[Function]}
          onHide={[Function]}
          onMount={[Function]}
          onShow={[Function]}
          onShown={[Function]}
          position="right"
          shouldClose={[Function]}
          zIndex={9999}
        >
          <React.Fragment>
            <Button
              variant="plain"
            >
              <InfoAltIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
                title={null}
              />
            </Button>
            e.g. http.status_code=200 error=true
          </React.Fragment>
        </Popover>
      </React.Fragment>
    }
    label="Tags"
  >
    <TextInput
      aria-label="tagsJaegerTraces"
      className="f6vm0mb"
      isDisabled={false}
      isReadOnly={false}
      isRequired={false}
      isValid={true}
      onChange={[MockFunction]}
      type="text"
      value=""
    />
  </FormGroup>
</Form>
`;
